runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -2384,6 +2384,37 @@ TR_J9ServerVM::getVMIndexOffset()
    vmInfo->_vmindexOffset = std::get<0>(stream->read<UDATA>());
    return vmInfo->_vmindexOffset;
    }
+
+TR::KnownObjectTable::Index
+TR_J9ServerVM::getMethodHandleTableEntryIndex(TR::Compilation *comp, TR::KnownObjectTable::Index vhIndex, TR::KnownObjectTable::Index adIndex)
+   {
+   TR::KnownObjectTable *knot = comp->getKnownObjectTable();
+   if (!knot) return TR::KnownObjectTable::UNKNOWN;
+
+   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
+   stream->write(JITServer::MessageType::VM_getMethodHandleTableEntryIndex, vhIndex, adIndex);
+   auto recv = stream->read<TR::KnownObjectTable::Index, uintptr_t *>();
+
+   TR::KnownObjectTable::Index mhIndex = std::get<0>(recv);
+   knot->updateKnownObjectTableAtServer(mhIndex, std::get<1>(recv));
+   return mhIndex;
+   }
+
+TR::KnownObjectTable::Index
+TR_J9ServerVM::getDirectVarHandleTargetIndex(TR::Compilation *comp, TR::KnownObjectTable::Index vhIndex)
```

Comment:
I don't think we need this method at all. Looking at `TR_J9VMBase::getDirectVarHandleTargetIndex` I see it does two main things:
- It calls `getSystemClassFromClassName()` which is already overridden at the server
- It calls `getObjectClassFromKnownObjectIndex(comp, vhIndex)` which is already overridden at the server 

Moreover, as far as I can tell, it does not create a new index in the `knot`, so we don't need to call `knot->updateKnownObjectTableAtServer(directVHIndex, std::get<1>(recv));`